modules/tensor_mechanics/test/tests/beam/constraints/frictional_constraint.i

```diff
@@ -0,0 +1,302 @@
+# y-dir displacement of the long beam is still 3.537e-3. However, the y-dir
+# displacement of the short beam increases until the force exceeds the
+# frictional capacity which in this case is 0.05 and then remains constant
+# after that point.
+
+[Mesh]
+  file = beam_cons_patch.e
+  displacements = 'disp_x disp_y disp_z'
+[]
+
+[Variables]
+  [./disp_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./disp_z]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./rot_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./rot_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./rot_z]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[BCs]
+  [./fixx1]
+    type = DirichletBC
+    variable = disp_x
+    boundary = '1001 1003'
+    value = 0.0
+  [../]
+  [./fixy1]
+    type = DirichletBC
+    variable = disp_y
+    boundary = '1001 1003'
+    value = 0.0
+  [../]
+  [./fixz1]
+    type = DirichletBC
+    variable = disp_z
+    boundary = '1001 1003'
+    value = 0.0
+  [../]
+  [./fixr1]
+    type = DirichletBC
+    variable = rot_x
+    boundary = '1001 1003'
+    value = 0.0
+  [../]
+  [./fixr2]
+    type = DirichletBC
+    variable = rot_y
+    boundary = '1001 1003'
+    value = 0.0
+  [../]
+  [./fixr3]
+    type = DirichletBC
+    variable = rot_z
+    boundary = '1001 1003'
+    value = 0.0
+  [../]
+[]
+
+[Constraints]
+  [./tie_y_fuel]
+    type = NodalFrictionalConstraint
+    normal_force = 1.0
+    tangential_penalty = 1.2e5
+    friction_coefficient = 0.05
+    boundary = 1005
+    slave = 1004
+    variable = disp_y
+  [../]
+  [./tie_x_fuel]
+    type = NodalStickConstraint
+    penalty = 1.2e14
+    boundary = 1005
+    slave = 1004
+    variable = disp_x
+  [../]
+  [./tie_z_fuel]
+    type = NodalStickConstraint
+    penalty = 1.2e14
+    boundary = 1005
+    slave = 1004
+    variable = disp_z
+  [../]
+  [./tie_rot_y_fuel]
+    type = NodalStickConstraint
+    penalty = 1.2e14
+    boundary = 1005
+    slave = 1004
+    variable = rot_y
+  [../]
+  [./tie_rot_x_fuel]
+    type = NodalStickConstraint
+    penalty = 1.2e14
+    boundary = 1005
+    slave = 1004
+    variable = rot_x
+  [../]
+  [./tie_rot_z_fuel]
+    type = NodalStickConstraint
+    penalty = 1.2e14
+    boundary = 1005
+    slave = 1004
+    variable = rot_z
+  [../]
+[]
+
+[Functions]
+  [./force_loading]
+    type = PiecewiseLinear
+    x = '0.0 5.0'
+    y = '0.0 50000.0'
+  [../]
+[]
+
+[NodalKernels]
+  [./force_x2]
+    type = UserForcingFunctionNodalKernel
+    variable = disp_y
+    boundary = '1004'
+    function = force_loading
+  [../]
+[]
+
+[Preconditioning]
+  [./smp]
+    type = SMP
+    full = true
+  [../]
+[]
+[Executioner]
+  type = Transient
+  solve_type = PJFNK
+  line_search = 'none'
+  nl_max_its = 15
+  nl_rel_tol = 1e-10
+  nl_abs_tol = 1e-8
+
+  dt = 1
+  dtmin = 1
+  end_time = 5
+[]
+
+[Kernels]
+  [./solid_disp_x]
```

Comment:
Isn't there an Action that sets this stuff up?  LineElementAction or something? If so, use it.